pkg/healthchecker/health_checker_windows.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthchecker
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/node-problem-detector/cmd/healthchecker/options"
+	"k8s.io/node-problem-detector/pkg/healthchecker/types"
+	"k8s.io/node-problem-detector/pkg/util"
+)
+
+type windowsHealthChecker struct {
+	component       string
+	service         string
+	enableRepair    bool
+	healthCheckFunc func() (bool, error)
+	// The repair is "best-effort" and ignores the error from the underlying actions.
+	// The powershell commands to restart the process will stop and start the service when down.
+	repairFunc         func()
+	uptimeFunc         func() (time.Duration, error)
+	crictlPath         string
+	healthCheckTimeout time.Duration
+	coolDownTime       time.Duration
+	logPatternsToCheck map[string]int
+}
+
+// NewWindowsHealthChecker returns a new health checker configured with the given options.
+func NewWindowsHealthChecker(hco *options.WindowsHealthCheckerOptions) (types.HealthChecker, error) {
```

Comment:
I'd expect this to be NewHealthChecker so that you trigger a compile error if both files are included at build time.